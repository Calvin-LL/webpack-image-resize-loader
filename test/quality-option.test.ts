import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe('"quality" option', () => {
  it("should work with 80", async () => {
    const compiler = getCompiler({
      width: 10,
      quality: 80,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  it("should work with 1", async () => {
    const compiler = getCompiler({
      width: 10,
      quality: 1,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });
});
