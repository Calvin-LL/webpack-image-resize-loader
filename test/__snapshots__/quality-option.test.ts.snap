// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"quality" option should work with 80: result 1`] = `"e9cac5dc1b3f4b537ff58229490ade0c.jpg"`;
